### ISSUE TITLE
Crash in -[MGLMapView willResignActive:] 

### ISSUE THREAD

**Issue by marcelofabri** [mapbox/mapbox-gl-native#14732]
### Steps to reproduce

Unable to reproduce, found via crash reporting.

### Expected behavior

It shouldn't crash.

### Actual behavior

```
Crashed: com.apple.main-thread
0  libGPUSupportMercury.dylib     0x1abd3afe4 gpus_ReturnNotPermittedKillClient
1  AGXGLDriver                    0x1b0284ec8 (Missing)
2  libGPUSupportMercury.dylib     0x1abd3bfac gpusSubmitDataBuffers
3  AGXGLDriver                    0x1b02863f4 (Missing)
4  AGXGLDriver                    0x1b0281af0 (Missing)
5  AGXGLDriver                    0x1b026a3cc (Missing)
6  GLEngine                       0x1b1361e38 glReadPixels_Exec
7  GLKit                          0x19ce7ec18 -[GLKView snapshot] + 248
8  Mapbox                         0x1066d1440 -[MGLMapView willResignActive:] (MGLMapView.mm:1487)
9  CoreFoundation                 0x18fed15bc __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 20
10 CoreFoundation                 0x18fed1588 ___CFXRegistrationPost_block_invoke + 64
11 CoreFoundation                 0x18fed0a7c _CFXRegistrationPost + 392
12 CoreFoundation                 0x18fed0728 ___CFXNotificationPost_block_invoke + 96
13 CoreFoundation                 0x18fe4a524 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 1496
14 CoreFoundation                 0x18fed01d8 _CFXNotificationPost + 696
15 Foundation                     0x1908b8814 -[NSNotificationCenter postNotificationName:object:userInfo:] + 68
16 UIKitCore                      0x1bc355848 -[UIApplication _deactivateForReason:notify:] + 932
17 UIKitCore                      0x1bbc230cc -[__UICanvasLifecycleMonitor_Compatability deactivateEventsOnly:withContext:forceExit:completion:] + 660
18 UIKitCore                      0x1bbc20a14 __82-[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:]_block_invoke + 824
19 UIKitCore                      0x1bbc2068c -[_UIApplicationCanvas _transitionLifecycleStateWithTransitionContext:completion:] + 428
20 UIKitCore                      0x1bbc251cc __125-[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:]_block_invoke + 220
21 UIKitCore                      0x1bbc25fb0 _performActionsWithDelayForTransitionContext + 112
22 UIKitCore                      0x1bbc25084 -[_UICanvasLifecycleSettingsDiffAction performActionsForCanvas:withUpdatedScene:settingsDiff:fromSettings:transitionContext:] + 244
23 UIKitCore                      0x1bbc29d84 -[_UICanvas scene:didUpdateWithDiff:transitionContext:completion:] + 360
24 UIKitCore                      0x1bbf593b8 -[UIApplicationSceneClientAgent scene:handleEvent:withCompletion:] + 464
25 FrontBoardServices             0x1928db920 __80-[FBSSceneImpl updater:didUpdateSettings:withDiff:transitionContext:completion:]_block_invoke_3 + 224
26 libdispatch.dylib              0x18f9a17d4 _dispatch_client_callout + 16
27 libdispatch.dylib              0x18f9465d8 _dispatch_block_invoke_direct$VARIANT$mp + 224
28 FrontBoardServices             0x1929150bc __FBSSERIALQUEUE_IS_CALLING_OUT_TO_A_BLOCK__ + 40
29 FrontBoardServices             0x192914d58 -[FBSSerialQueue _performNext] + 408
30 FrontBoardServices             0x192915310 -[FBSSerialQueue _performNextFromRunLoopSource] + 52
31 CoreFoundation                 0x18fef22bc __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 24
32 CoreFoundation                 0x18fef223c __CFRunLoopDoSource0 + 88
33 CoreFoundation                 0x18fef1b24 __CFRunLoopDoSources0 + 176
34 CoreFoundation                 0x18feeca60 __CFRunLoopRun + 1004
35 CoreFoundation                 0x18feec354 CFRunLoopRunSpecific + 436
36 GraphicsServices               0x1920ec79c GSEventRunModal + 104
37 UIKitCore                      0x1bc35fb68 UIApplicationMain + 212
38 InstaShopper                   0x10117a194 main (<compiler-generated>)
39 libdyld.dylib                  0x18f9b28e0 start + 4
```

### Configuration

**Mapbox SDK versions:** 4.10
**iOS/macOS versions:** iOS 11 and 12
**Device/simulator models:** iPhone 6, 6 Plus and 7 Plus 
**Xcode version:** 10.2

**Comment by julianrex** [comment]
Closing as this was addressed in 4.11.0 release (see comment here https://github.com/mapbox/mapbox-gl-native/issues/13821#issuecomment-493966923)

**Comment by ansarikhurshid786** [comment]
I also getting this crashing report with same errror. @julianrex url link is not valid.
